packages/esm-patient-medications-app/src/add-drug-order/add-drug-order.workspace.tsx

```diff
@@ -67,8 +78,12 @@ export default function AddDrugOrderWorkspace({
         onWorkspaceClose: () => launchWorkspace('order-basket'),
       });
     },
-    [orders, setOrders, closeWorkspaceWithSavedChanges, session.currentProvider.uuid],
+    [orders, setOrders, closeWorkspaceWithSavedChanges, session.currentProvider, t],
   );
+  //makes the current provider to hydrate first before the session continues
+  if (!session.currentProvider) {
+    return <div className={styles.loading}>{t('loadingProvider', 'Loading provider information...')}</div>;
+  }
```

Comment:
What guarantees that `session.currentProvider` will ever return a result? What happens when this is never materializes?